packages/jest-haste-map/src/index.ts

```diff
@@ -788,12 +779,21 @@ class HasteMap extends EventEmitter {
     let mustCopy = true;
 
     const createWatcher = (root: Config.Path): Promise<Watcher> => {
-      // @ts-expect-error: TODO how? "Cannot use 'new' with an expression whose type lacks a call or construct signature."
-      const watcher = new Watcher(root, {
-        dot: false,
-        glob: extensions.map(extension => '**/*.' + extension),
-        ignored: ignorePattern,
-      });
+      const useWatchman = canUseWatchman && this._options.useWatchman;
+      const patterns = extensions.map(extension => '**/*.' + extension);
+      // Prefer Watchman over Chokidar
+      const watcher = useWatchman
+        ? new WatchmanWatcher(root, {
+            dot: false,
+            glob: patterns,
+            ignored: ignorePattern,
+          })
+        : chokidarWatch(patterns, {
```

Comment:
Surprising but .. they expose almost the same API? :O

Reply:
I dunno if the ignore pattern works, should probably verify